### DISCUSSION TITLE
What is the support of mmaction2 for data stored remotely on S3 or Hadoop Filesystem?

### DISCUSSION THREAD

**Discussion by yxchng** [open-mmlab/mmaction2#636] [General]
Based on the current implementation, https://github.com/open-mmlab/mmcv/blob/c9f96855b06a05170cf8fd658eed21092e26ee47/mmcv/fileio/file_client.py, it seems like videos are downloaded all the time at the start of every batch, wouldn't it be very slow?

**Comment by innerlee** [comment]
It is assumed that s3 is fast.

If not, then consider to switch the io client.

**Comment by yxchng** [comment]
@innerlee Do you have any experience training over data stored in s3-like services like Aliyun OSS? Do you see any slowdown in speed in such use case? I think the speed is mainly dependant on your network speed, not really the client.

**Comment by innerlee** [comment]
If the gpu server is also located at Aliyun, and if they have high-speed internal networks such as infiniband (and if they do not charge you much), then you can directly train from s3. This is my speculation.
